<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement &amp; Alexandre Segers" />


<title>Experimenteel Design II: replicatie en power - KPNA2</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>


<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSLS21</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-intro.html">1. Introduction</a>
    </li>
    <li>
      <a href="02-concepts.html">2. Concepts</a>
    </li>
    <li>
      <a href="03-experimentalDesign.html">3. Experimental Design I</a>
    </li>
    <li>
      <a href="04-dataExploration.html">4. Data Exploration</a>
    </li>
    <li>
      <a href="05-statisticalInference.html">5.1 Statistical Inference 1</a>
    </li>
    <li>
      <a href="05-statisticalInference-twosampleT.html">5.2 Statistical Inference 2</a>
    </li>
    <li>
      <a href="06-linearRegression.html">6. Linear Regression</a>
    </li>
    <li>
      <a href="07-Anova.html">7. Analysis of Variance</a>
    </li>
    <li>
      <a href="08-MultipleRegression.html">8. Multiple Regression</a>
    </li>
    <li>
      <a href="08-MultipleRegression-SpecificDesigns.html">8.2 Multiple Regression: Factorial Designs</a>
    </li>
    <li>
      <a href="08-experimentalDesignII.html">8.3. Experimental Design II</a>
    </li>
    <li>
      <a href="08-experimentalDesignIII.html">8.4. Experimental Design II</a>
    </li>
    <li>
      <a href="09-NonparametericStatistics-WilcoxonMannWithney.html">9.1. Nonparametric Statistics 1</a>
    </li>
    <li>
      <a href="09-NonparametericStatistics-KruskalWallis.html">9.2. Nonparametric Statistics 2</a>
    </li>
    <li>
      <a href="10-categoricalDataAnalysis.html">10. Categorical Data Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Labs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04_exercises.html">4. Data Exploration</a>
    </li>
    <li>
      <a href="05_exercises.html">5. Statistical inference</a>
    </li>
    <li>
      <a href="06_exercises.html">6. Simple linear regression</a>
    </li>
    <li>
      <a href="07_exercises.html">7. ANOVA</a>
    </li>
    <li>
      <a href="08_exercises.html">8. Multiple linear regression</a>
    </li>
    <li>
      <a href="09_exercises.html">9. Non-parametric statistics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/PSLS21">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Experimenteel Design II: replicatie en power - KPNA2</h1>
<h4 class="author">Lieven Clement &amp; Alexandre Segers</h4>
<h4 class="date">statOmics, Ghent University (<a href="https://statomics.github.io" class="uri">https://statomics.github.io</a>)</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>## ✔ tibble  3.1.4     ✔ dplyr   1.0.7
## ✔ tidyr   1.1.4     ✔ stringr 1.4.0
## ✔ readr   2.0.2     ✔ forcats 0.5.1
## ✔ purrr   0.3.4</code></pre>
<pre><code>## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()</code></pre>
<div id="background" class="section level1">
<h1><span class="header-section-number">1</span> Background</h1>
<p>Histologic grade in breast cancer provides clinically important prognostic information. Researchers examined whether histologic grade was associated with gene expression profiles of breast cancers and whether such profiles could be used to improve histologic grading. In this tutorial we will assess the association between histologic grade and the expression of the KPNA2 gene that is known to be associated with poor BC prognosis. The patients, however, do not only differ in the histologic grade, but also on their lymph node status. The lymph nodes were not affected (0) or chirugically removed (1).</p>
<ul>
<li>Redo data analysis (you can copy the results of the tutorial on multiple linear regression)</li>
<li>What is the power to pick up each of the contrasts when their real effect sizes would be equal to the effect sizes we observed in the study?</li>
<li>How does the power evolves if we have 2 upto 10 repeats for each factor combination of grade and node when their real effect sizes would be equal to the ones we observed in the study?</li>
<li>What is the power to pick up each of the contrasts when the FC for grade for patients with unaffected lymf nodes equals 1.5 (<span class="math inline">\(\beta_g = log2(1.5)\)</span>)?</li>
</ul>
</div>
<div id="data-analysis" class="section level1">
<h1><span class="header-section-number">2</span> Data analysis</h1>
<div id="import-kpna2-data-in-r" class="section level2">
<h2><span class="header-section-number">2.1</span> Import KPNA2 data in R</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>kpna2 &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/SGA21/master/data/kpna2.txt&quot;</span>,<span class="dt">header=</span><span class="ot">TRUE</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a>kpna2</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["grade"],"name":[1],"type":["int"],"align":["right"]},{"label":["node"],"name":[2],"type":["int"],"align":["right"]},{"label":["gene"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"3","2":"1","3":"367.8179"},{"1":"3","2":"1","3":"590.3576"},{"1":"1","2":"1","3":"346.6583"},{"1":"1","2":"1","3":"258.4455"},{"1":"1","2":"0","3":"153.8416"},{"1":"3","2":"0","3":"643.6799"},{"1":"3","2":"1","3":"817.8558"},{"1":"1","2":"1","3":"329.4113"},{"1":"3","2":"0","3":"746.4951"},{"1":"3","2":"0","3":"380.0940"},{"1":"1","2":"0","3":"205.2980"},{"1":"3","2":"0","3":"703.5070"},{"1":"1","2":"0","3":"223.5533"},{"1":"1","2":"0","3":"186.6673"},{"1":"1","2":"0","3":"165.5948"},{"1":"3","2":"1","3":"439.0382"},{"1":"1","2":"1","3":"252.0597"},{"1":"3","2":"0","3":"495.8720"},{"1":"1","2":"1","3":"286.7907"},{"1":"3","2":"1","3":"552.1972"},{"1":"1","2":"1","3":"233.5769"},{"1":"3","2":"0","3":"521.4048"},{"1":"3","2":"1","3":"474.2651"},{"1":"1","2":"0","3":"148.1059"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Because histolic grade and lymph node status are both categorical variables, we model them both as factors.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>kpna2<span class="op">$</span>grade &lt;-<span class="st"> </span><span class="kw">as.factor</span>(kpna2<span class="op">$</span>grade)</span>
<span id="cb6-2"><a href="#cb6-2"></a>kpna2<span class="op">$</span>node &lt;-<span class="st"> </span><span class="kw">as.factor</span>(kpna2<span class="op">$</span>node)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>kpna2 <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>node<span class="op">:</span>grade,<span class="dt">y=</span>gene,<span class="dt">fill=</span>node<span class="op">:</span>grade)) <span class="op">+</span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.shape =</span> <span class="ot">NA</span>) <span class="op">+</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="st">  </span><span class="kw">geom_jitter</span>()</span></code></pre></div>
<p><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>As discussed in a previous exercise, it seems that there is both an effect of histologic grade and lymph node status on the gene expression. There also seems to be a different effect of lymph node status on the gene expression for the different histologic grades.</p>
<p>As we saw before, we can model this with a model that contains both histologic grade, lymph node status and the interaction term between both. When checking the linear model assumptions, we see that the variance is not equal. Therefore we model the gene expression with a log2-transformation, which makes that all the assumptions of the linear model are satisfied.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co">#Model with main effects for histological grade and node and grade x node interaction</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>fit &lt;-<span class="st"> </span><span class="kw">lm</span>(gene<span class="op">~</span>grade<span class="op">*</span>node,<span class="dt">data=</span>kpna2)</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="kw">summary</span>(fit)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = gene ~ grade * node, data = kpna2)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -201.748  -53.294   -6.308   46.216  277.601 
## 
## Coefficients:
##              Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)    180.51      44.37   4.068   0.0006 ***
## grade3         401.33      62.75   6.396 3.07e-06 ***
## node1          103.98      62.75   1.657   0.1131    
## grade3:node1  -145.57      88.74  -1.640   0.1166    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 108.7 on 20 degrees of freedom
## Multiple R-squared:  0.7437, Adjusted R-squared:  0.7052 
## F-statistic: 19.34 on 3 and 20 DF,  p-value: 3.971e-06</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">plot</span>(fit)</span></code></pre></div>
<p><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-5-1.png" width="672" /><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-5-2.png" width="672" /><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-5-3.png" width="672" /><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-5-4.png" width="672" /></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>fit &lt;-<span class="st"> </span><span class="kw">lm</span>(gene <span class="op">%&gt;%</span><span class="st"> </span>log2<span class="op">~</span>grade<span class="op">*</span>node,<span class="dt">data=</span>kpna2)</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw">plot</span>(fit)</span></code></pre></div>
<p><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-6-1.png" width="672" /><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-6-2.png" width="672" /><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-6-3.png" width="672" /><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-6-4.png" width="672" /></p>
<p>When checking the significance of the interaction term, we see that it is significant on the 5% significance level. We therefore keep the full model.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="kw">library</span>(car)</span></code></pre></div>
<pre><code>## Loading required package: carData</code></pre>
<pre><code>## 
## Attaching package: &#39;car&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     recode</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     some</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="kw">Anova</span>(fit,<span class="dt">type=</span><span class="st">&quot;III&quot;</span>)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Sum Sq"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Df"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["F value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Pr(>F)"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"335.6676323","2":"1","3":"3351.610907","4":"9.530118e-24","_rn_":"(Intercept)"},{"1":"8.3420526","2":"1","3":"83.294639","4":"1.437549e-08","_rn_":"grade"},{"1":"1.2978311","2":"1","3":"12.958726","4":"1.789077e-03","_rn_":"node"},{"1":"0.9003583","2":"1","3":"8.989996","4":"7.102512e-03","_rn_":"grade:node"},{"1":"2.0030227","2":"20","3":"NA","4":"NA","_rn_":"Residuals"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>As we are dealing with a factorial design, we can calculate the mean gene expression for each group by the following parameter summations.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>ExploreModelMatrix<span class="op">::</span><span class="kw">VisualizeDesign</span>(kpna2,<span class="op">~</span>grade<span class="op">*</span>node)<span class="op">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>The researchers want to know the power for testing following hypotheses (remark that we will have to adjust for multiple testing):</p>
<ul>
<li>Log fold change between histologic grade 3 and histologic grade 1 for patients with unaffected lymph nodes (=0).</li>
</ul>
<p><span class="math display">\[H_0: \log_2{FC}_{g3n0-g1n0} = \beta_{g3} = 0\]</span></p>
<ul>
<li>Log fold change between histologic grade 3 and histologic grade 1 for patients with removed lymph nodes (=1).</li>
</ul>
<p><span class="math display">\[H_0: \log_2{FC}_{g3n1-g1n1} = \beta_{g3} + \beta_{g3n1} = 0\]</span></p>
<ul>
<li>Log fold change between unaffected and removed lymph nodes for patients of histologic grade 1.</li>
</ul>
<p><span class="math display">\[H_0: \log_2{FC}_{g1n1-g1n0} = \beta_{n1} = 0\]</span></p>
<ul>
<li>Log fold change between unaffected and removed lymph nodes for patients of histologic grade 3.</li>
</ul>
<p><span class="math display">\[H_0: \log_2{FC}_{g3n1-g3n0} = \beta_{n1} + \beta_{g3n1} = 0\]</span></p>
<ul>
<li>Difference in log fold change between patients of histological grade 3 and histological grade 1 with removed lymph nodes and log fold change between patients between patients of histological grade 3 and histological grade 1 with unaffected lymph nodes.</li>
</ul>
<p><span class="math display">\[H_0: \log_2{FC}_{g3n1-g1n1} - \log_2{FC}_{g3n0-g1n0} = \beta_{g3n1} = 0\]</span> which is an equivalent hypotheses with <span class="math display">\[H_0: \log_2{FC}_{g3n1-g3n0} - \log_2{FC}_{g1n1-g1n0} = \beta_{g3n1} = 0\]</span></p>
<p>We can test this using multcomp, which controls for multiple testing.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="kw">library</span>(multcomp)</span></code></pre></div>
<pre><code>## Loading required package: mvtnorm</code></pre>
<pre><code>## Loading required package: survival</code></pre>
<pre><code>## Loading required package: TH.data</code></pre>
<pre><code>## Loading required package: MASS</code></pre>
<pre><code>## 
## Attaching package: &#39;MASS&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     select</code></pre>
<pre><code>## 
## Attaching package: &#39;TH.data&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MASS&#39;:
## 
##     geyser</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>mcp &lt;-<span class="st"> </span><span class="kw">glht</span>(fit,<span class="dt">linfct =</span> <span class="kw">c</span>(<span class="st">&quot;grade3 = 0&quot;</span>,<span class="st">&quot;grade3 + grade3:node1 = 0&quot;</span>, <span class="st">&quot;node1 = 0&quot;</span>,<span class="st">&quot;node1 + grade3:node1 = 0&quot;</span>, <span class="st">&quot;grade3:node1 = 0&quot;</span>))</span>
<span id="cb29-2"><a href="#cb29-2"></a><span class="kw">summary</span>(mcp)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: lm(formula = gene %&gt;% log2 ~ grade * node, data = kpna2)
## 
## Linear Hypotheses:
##                            Estimate Std. Error t value Pr(&gt;|t|)    
## grade3 == 0                  1.6675     0.1827   9.127  &lt; 0.001 ***
## grade3 + grade3:node1 == 0   0.8928     0.1827   4.886  &lt; 0.001 ***
## node1 == 0                   0.6577     0.1827   3.600  0.00714 ** 
## node1 + grade3:node1 == 0   -0.1170     0.1827  -0.640  0.89815    
## grade3:node1 == 0           -0.7748     0.2584  -2.998  0.02683 *  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<p>We get a significant p-value for the first, second, third and fifth hypothesis. The fourth hypothesis is not significant at the overall 5% significance level.</p>
</div>
</div>
<div id="power-of-the-tests-for-each-of-the-contrasts" class="section level1">
<h1><span class="header-section-number">3</span> Power of the tests for each of the contrasts</h1>
<div id="simulation-function" class="section level2">
<h2><span class="header-section-number">3.1</span> Simulation function</h2>
<p>Function to simulate data similar to that of our experiment under our model assumptions.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>simFastMultipleContrasts &lt;-<span class="st"> </span><span class="cf">function</span>(form, data, betas, sd, contrasts, <span class="dt">alpha =</span> <span class="fl">.05</span>, <span class="dt">nSim =</span> <span class="dv">10000</span>, <span class="dt">adjust =</span> <span class="st">&quot;holm&quot;</span>)</span>
<span id="cb31-2"><a href="#cb31-2"></a>{</span>
<span id="cb31-3"><a href="#cb31-3"></a>    ySim &lt;-<span class="st"> </span><span class="kw">rnorm</span>(<span class="kw">nrow</span>(data)<span class="op">*</span>nSim,<span class="dt">sd=</span>sd)</span>
<span id="cb31-4"><a href="#cb31-4"></a>    <span class="kw">dim</span>(ySim) &lt;-<span class="kw">c</span>(<span class="kw">nrow</span>(data),nSim)</span>
<span id="cb31-5"><a href="#cb31-5"></a>    design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(form, data)</span>
<span id="cb31-6"><a href="#cb31-6"></a>    ySim &lt;-<span class="st"> </span>ySim <span class="op">+</span><span class="st"> </span><span class="kw">c</span>(design <span class="op">%*%</span>betas)</span>
<span id="cb31-7"><a href="#cb31-7"></a>    ySim &lt;-<span class="st"> </span><span class="kw">t</span>(ySim)</span>
<span id="cb31-8"><a href="#cb31-8"></a>  </span>
<span id="cb31-9"><a href="#cb31-9"></a>    <span class="co">### Fitting</span></span>
<span id="cb31-10"><a href="#cb31-10"></a>    fitAll &lt;-<span class="st"> </span>limma<span class="op">::</span><span class="kw">lmFit</span>(ySim,design)</span>
<span id="cb31-11"><a href="#cb31-11"></a>  </span>
<span id="cb31-12"><a href="#cb31-12"></a>    <span class="co">### Inference</span></span>
<span id="cb31-13"><a href="#cb31-13"></a>    varUnscaled &lt;-<span class="st"> </span><span class="kw">t</span>(contrasts)<span class="op">%*%</span>fitAll<span class="op">$</span>cov.coefficients<span class="op">%*%</span>contrasts</span>
<span id="cb31-14"><a href="#cb31-14"></a>    contrasts &lt;-<span class="st"> </span>fitAll<span class="op">$</span>coefficients <span class="op">%*%</span>contrasts</span>
<span id="cb31-15"><a href="#cb31-15"></a>    seContrasts &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">diag</span>(varUnscaled)<span class="op">^</span>.<span class="dv">5</span>,<span class="dt">nrow=</span>nSim,<span class="dt">ncol=</span><span class="dv">5</span>,<span class="dt">byrow =</span> <span class="ot">TRUE</span>)<span class="op">*</span>fitAll<span class="op">$</span>sigma</span>
<span id="cb31-16"><a href="#cb31-16"></a>    tstats &lt;-<span class="st"> </span>contrasts<span class="op">/</span>seContrasts</span>
<span id="cb31-17"><a href="#cb31-17"></a>    pvals &lt;-<span class="st"> </span><span class="kw">pt</span>(<span class="kw">abs</span>(tstats),fitAll<span class="op">$</span>df.residual,<span class="dt">lower.tail =</span> <span class="ot">FALSE</span>)<span class="op">*</span><span class="dv">2</span></span>
<span id="cb31-18"><a href="#cb31-18"></a>    pvals &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(pvals, <span class="dv">1</span>, p.adjust, <span class="dt">method =</span> adjust))</span>
<span id="cb31-19"><a href="#cb31-19"></a>    </span>
<span id="cb31-20"><a href="#cb31-20"></a>    <span class="kw">return</span>(<span class="kw">colMeans</span>(pvals <span class="op">&lt;</span><span class="st"> </span>alpha))</span>
<span id="cb31-21"><a href="#cb31-21"></a>}</span></code></pre></div>
</div>
<div id="power-of-current-experiment" class="section level2">
<h2><span class="header-section-number">3.2</span> power of current experiment</h2>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>nSim &lt;-<span class="st"> </span><span class="dv">20000</span></span>
<span id="cb32-2"><a href="#cb32-2"></a>betas &lt;-<span class="st"> </span>fit<span class="op">$</span>coefficients</span>
<span id="cb32-3"><a href="#cb32-3"></a>sd &lt;-<span class="st"> </span><span class="kw">sigma</span>(fit)</span>
<span id="cb32-4"><a href="#cb32-4"></a></span>
<span id="cb32-5"><a href="#cb32-5"></a>contrasts &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>,<span class="dt">nrow=</span><span class="dv">4</span>,<span class="dt">ncol=</span><span class="dv">5</span>)</span>
<span id="cb32-6"><a href="#cb32-6"></a><span class="kw">rownames</span>(contrasts) &lt;-<span class="st"> </span><span class="kw">names</span>(fit<span class="op">$</span>coefficients)</span>
<span id="cb32-7"><a href="#cb32-7"></a><span class="kw">colnames</span>(contrasts) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;graden0&quot;</span>,<span class="st">&quot;graden1&quot;</span>,<span class="st">&quot;nodeg1&quot;</span>,<span class="st">&quot;nodeg3&quot;</span>,<span class="st">&quot;interaction&quot;</span>)</span>
<span id="cb32-8"><a href="#cb32-8"></a>contrasts[<span class="dv">2</span>,<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb32-9"><a href="#cb32-9"></a>contrasts[<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">4</span>),<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb32-10"><a href="#cb32-10"></a>contrasts[<span class="dv">3</span>,<span class="dv">3</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb32-11"><a href="#cb32-11"></a>contrasts[<span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>),<span class="dv">4</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb32-12"><a href="#cb32-12"></a>contrasts[<span class="dv">4</span>,<span class="dv">5</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb32-13"><a href="#cb32-13"></a></span>
<span id="cb32-14"><a href="#cb32-14"></a>form &lt;-<span class="st"> </span><span class="er">~</span><span class="st"> </span>grade<span class="op">*</span>node</span>
<span id="cb32-15"><a href="#cb32-15"></a></span>
<span id="cb32-16"><a href="#cb32-16"></a>power1 &lt;-<span class="st"> </span><span class="kw">simFastMultipleContrasts</span>(form, kpna2, betas, sd, contrasts, <span class="dt">nSim =</span> nSim)</span>
<span id="cb32-17"><a href="#cb32-17"></a>power1</span></code></pre></div>
<pre><code>##     graden0     graden1      nodeg1      nodeg3 interaction 
##     1.00000     0.98950     0.84345     0.08855     0.70390</code></pre>
<p>We observe large powers for all contrasts, exept for contrast nodeg3, which has a small effect size.</p>
</div>
<div id="power-for-increasing-sample-size" class="section level2">
<h2><span class="header-section-number">3.3</span> Power for increasing sample size</h2>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>nSim &lt;-<span class="st"> </span><span class="dv">20000</span></span>
<span id="cb34-2"><a href="#cb34-2"></a>betas &lt;-<span class="st"> </span>fit<span class="op">$</span>coefficients</span>
<span id="cb34-3"><a href="#cb34-3"></a>sd &lt;-<span class="st"> </span><span class="kw">sigma</span>(fit)</span>
<span id="cb34-4"><a href="#cb34-4"></a></span>
<span id="cb34-5"><a href="#cb34-5"></a>contrasts &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>,<span class="dt">nrow=</span><span class="dv">4</span>,<span class="dt">ncol=</span><span class="dv">5</span>)</span>
<span id="cb34-6"><a href="#cb34-6"></a><span class="kw">rownames</span>(contrasts) &lt;-<span class="st"> </span><span class="kw">names</span>(fit<span class="op">$</span>coefficients)</span>
<span id="cb34-7"><a href="#cb34-7"></a><span class="kw">colnames</span>(contrasts) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;graden0&quot;</span>,<span class="st">&quot;graden1&quot;</span>,<span class="st">&quot;nodeg1&quot;</span>,<span class="st">&quot;nodeg3&quot;</span>,<span class="st">&quot;interaction&quot;</span>)</span>
<span id="cb34-8"><a href="#cb34-8"></a>contrasts[<span class="dv">2</span>,<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb34-9"><a href="#cb34-9"></a>contrasts[<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">4</span>),<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb34-10"><a href="#cb34-10"></a>contrasts[<span class="dv">3</span>,<span class="dv">3</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb34-11"><a href="#cb34-11"></a>contrasts[<span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>),<span class="dv">4</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb34-12"><a href="#cb34-12"></a>contrasts[<span class="dv">4</span>,<span class="dv">5</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb34-13"><a href="#cb34-13"></a></span>
<span id="cb34-14"><a href="#cb34-14"></a>form &lt;-<span class="st"> </span><span class="er">~</span><span class="st"> </span>grade<span class="op">*</span>node</span>
<span id="cb34-15"><a href="#cb34-15"></a></span>
<span id="cb34-16"><a href="#cb34-16"></a>powers &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>,<span class="dt">nrow=</span><span class="dv">9</span>, <span class="dt">ncol=</span><span class="dv">6</span>)</span>
<span id="cb34-17"><a href="#cb34-17"></a><span class="kw">colnames</span>(powers) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;n&quot;</span>,<span class="kw">colnames</span>(contrasts))</span>
<span id="cb34-18"><a href="#cb34-18"></a>powers[,<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="dv">2</span><span class="op">:</span><span class="dv">10</span></span>
<span id="cb34-19"><a href="#cb34-19"></a></span>
<span id="cb34-20"><a href="#cb34-20"></a>dataAllComb &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">grade =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>),<span class="dt">each=</span><span class="dv">2</span>)<span class="op">%&gt;%</span><span class="st"> </span>as.factor, </span>
<span id="cb34-21"><a href="#cb34-21"></a>                          <span class="dt">node =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="dv">2</span>)<span class="op">%&gt;%</span>as.factor) </span>
<span id="cb34-22"><a href="#cb34-22"></a>  </span>
<span id="cb34-23"><a href="#cb34-23"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(powers))</span>
<span id="cb34-24"><a href="#cb34-24"></a>{  </span>
<span id="cb34-25"><a href="#cb34-25"></a>predData &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">grade =</span> <span class="kw">rep</span>(dataAllComb<span class="op">$</span>grade, powers[i,<span class="dv">1</span>]), </span>
<span id="cb34-26"><a href="#cb34-26"></a>                       <span class="dt">node =</span> <span class="kw">rep</span>(dataAllComb<span class="op">$</span>node, powers[i,<span class="dv">1</span>]))</span>
<span id="cb34-27"><a href="#cb34-27"></a>powers[i,<span class="op">-</span><span class="dv">1</span>] &lt;-<span class="st"> </span><span class="kw">simFastMultipleContrasts</span>(form, predData, betas, sd, contrasts, <span class="dt">nSim =</span> nSim)</span>
<span id="cb34-28"><a href="#cb34-28"></a>}</span>
<span id="cb34-29"><a href="#cb34-29"></a>powers</span></code></pre></div>
<pre><code>##        n graden0 graden1  nodeg1  nodeg3 interaction
##  [1,]  2 0.70380 0.26610 0.15375 0.03220     0.11600
##  [2,]  3 0.99195 0.64345 0.37620 0.04475     0.27510
##  [3,]  4 0.99995 0.87370 0.58670 0.05950     0.44960
##  [4,]  5 1.00000 0.95905 0.73745 0.07375     0.58930
##  [5,]  6 1.00000 0.98855 0.84000 0.08645     0.70320
##  [6,]  7 1.00000 0.99720 0.90780 0.09630     0.78675
##  [7,]  8 1.00000 0.99945 0.94485 0.10615     0.85270
##  [8,]  9 1.00000 0.99995 0.97040 0.11300     0.89760
##  [9,] 10 1.00000 1.00000 0.98325 0.12785     0.93285</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>powers <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb36-2"><a href="#cb36-2"></a><span class="st">  </span>as.data.frame <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb36-3"><a href="#cb36-3"></a><span class="st">  </span><span class="kw">gather</span>(contrast, power, <span class="op">-</span>n) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb36-4"><a href="#cb36-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(n,power,<span class="dt">color=</span>contrast)) <span class="op">+</span></span>
<span id="cb36-5"><a href="#cb36-5"></a><span class="st">  </span><span class="kw">geom_line</span>()</span></code></pre></div>
<p><img src="08_ExperimentalDesignII_3_kpna2_sol_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
<div id="power-when-fc-for-grade-in-patients-with-unaffected-lymph-nodes-equals-1.5" class="section level2">
<h2><span class="header-section-number">3.4</span> Power when FC for grade in patients with unaffected lymph nodes equals 1.5</h2>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>nSim &lt;-<span class="st"> </span><span class="dv">20000</span></span>
<span id="cb37-2"><a href="#cb37-2"></a>betas2 &lt;-<span class="st"> </span>fit<span class="op">$</span>coefficients</span>
<span id="cb37-3"><a href="#cb37-3"></a>betas2[<span class="st">&quot;grade3&quot;</span>] &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="fl">1.5</span>)</span>
<span id="cb37-4"><a href="#cb37-4"></a>sd &lt;-<span class="st"> </span><span class="kw">sigma</span>(fit)</span>
<span id="cb37-5"><a href="#cb37-5"></a></span>
<span id="cb37-6"><a href="#cb37-6"></a>contrasts &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>,<span class="dt">nrow=</span><span class="dv">4</span>,<span class="dt">ncol=</span><span class="dv">5</span>)</span>
<span id="cb37-7"><a href="#cb37-7"></a><span class="kw">rownames</span>(contrasts) &lt;-<span class="st"> </span><span class="kw">names</span>(fit<span class="op">$</span>coefficients)</span>
<span id="cb37-8"><a href="#cb37-8"></a><span class="kw">colnames</span>(contrasts) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;graden0&quot;</span>,<span class="st">&quot;graden1&quot;</span>,<span class="st">&quot;nodeg1&quot;</span>,<span class="st">&quot;nodeg3&quot;</span>,<span class="st">&quot;interaction&quot;</span>)</span>
<span id="cb37-9"><a href="#cb37-9"></a>contrasts[<span class="dv">2</span>,<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb37-10"><a href="#cb37-10"></a>contrasts[<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">4</span>),<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb37-11"><a href="#cb37-11"></a>contrasts[<span class="dv">3</span>,<span class="dv">3</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb37-12"><a href="#cb37-12"></a>contrasts[<span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>),<span class="dv">4</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb37-13"><a href="#cb37-13"></a>contrasts[<span class="dv">4</span>,<span class="dv">5</span>] &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb37-14"><a href="#cb37-14"></a></span>
<span id="cb37-15"><a href="#cb37-15"></a>form &lt;-<span class="st"> </span><span class="er">~</span><span class="st"> </span>grade<span class="op">*</span>node</span>
<span id="cb37-16"><a href="#cb37-16"></a></span>
<span id="cb37-17"><a href="#cb37-17"></a>power3 &lt;-<span class="st"> </span><span class="kw">simFastMultipleContrasts</span>(form, kpna2, betas2, sd, contrasts,<span class="dt">nSim =</span> nSim)</span>
<span id="cb37-18"><a href="#cb37-18"></a>power3</span></code></pre></div>
<pre><code>##     graden0     graden1      nodeg1      nodeg3 interaction 
##     0.69180     0.10220     0.78935     0.06050     0.62705</code></pre>
<p>We observe a drop in the power for all the contrasts!?</p>
<p>We only would expect to loose power for the contrasts assessing the grade effect because both fold changes have been decreased.</p>
<p>The reason why all powers are affected is because we also do multiple testing and the multiple testing using the pvalue correction using Holm’s method for a particular contrast is also depending on the order of significance of the tests within each simulation.</p>
<p>When we do both simulations without correcting for multiple testing (adjust = “none”) we see that changing the size of the fold change for grade for patients with unaffected lymph nodes only impacts the power for the contrasts for assessing the grade effect.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a>power1None &lt;-<span class="st"> </span><span class="kw">simFastMultipleContrasts</span>(form, kpna2, betas, sd, contrasts, <span class="dt">nSim =</span> nSim, <span class="dt">adjust =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb39-2"><a href="#cb39-2"></a>power3None &lt;-<span class="st"> </span><span class="kw">simFastMultipleContrasts</span>(form, kpna2, betas2, sd, contrasts, <span class="dt">nSim =</span> nSim, <span class="dt">adjust =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb39-3"><a href="#cb39-3"></a>power1None</span></code></pre></div>
<pre><code>##     graden0     graden1      nodeg1      nodeg3 interaction 
##     1.00000     0.99585     0.92755     0.08985     0.81210</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a>power3None</span></code></pre></div>
<pre><code>##     graden0     graden1      nodeg1      nodeg3 interaction 
##     0.86510     0.16725     0.93095     0.09265     0.81600</code></pre>
</div>
</div>

<div id="rmd-source-code">---
title: "Experimenteel Design II: replicatie en power - KPNA2"
author: "Lieven Clement & Alexandre Segers"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
  html_document:
    code_download: yes
    theme: cosmo
    toc: yes
    toc_float: yes
    highlight: tango
    number_sections: yes
---


<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(ggplot2)
library(tidyverse)
```

# Background

Histologic grade in breast cancer provides clinically important prognostic information. Researchers examined whether histologic grade was associated with gene expression profiles of breast cancers and whether such profiles could be used to improve histologic grading. In this tutorial we will assess the association between histologic grade and the expression of the KPNA2 gene that is known to be associated with poor BC prognosis.
The patients, however, do not only differ in the histologic grade, but also on their lymph node status. 
The lymph nodes were not affected (0) or chirugically removed (1).

- Redo data analysis (you can copy the results of the tutorial on multiple linear regression)
- What is the power to pick up each of the contrasts when their real effect sizes would be equal to the effect sizes we observed in the study? 
- How does the power evolves if we have 2 upto 10 repeats for each factor combination of grade and node when their real effect sizes would be equal to the ones we observed in the study? 
- What is the power to pick up each of the contrasts when the FC for grade for patients with unaffected lymf nodes equals 1.5 ($\beta_g = log2(1.5)$)? 


# Data analysis
## Import KPNA2 data in R
```{r}
kpna2 <- read.table("https://raw.githubusercontent.com/statOmics/SGA21/master/data/kpna2.txt",header=TRUE)
kpna2
```

Because histolic grade and lymph node status are both categorical variables, we model them both as factors.

```{r}
kpna2$grade <- as.factor(kpna2$grade)
kpna2$node <- as.factor(kpna2$node)

```

```{r}
kpna2 %>% 
  ggplot(aes(x=node:grade,y=gene,fill=node:grade)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter()
```

As discussed in a previous exercise, it seems that there is both an effect of histologic grade and lymph node status on the gene expression. There also seems to be a different effect of lymph node status on the gene expression for the different histologic grades.

As we saw before, we can model this with a model that contains both histologic grade, lymph node status and the interaction term between both. When checking the linear model assumptions, we see that the variance is not equal. Therefore we model the gene expression with a log2-transformation, which makes that all the assumptions of the linear model are satisfied.


```{r}
#Model with main effects for histological grade and node and grade x node interaction
fit <- lm(gene~grade*node,data=kpna2)
summary(fit)
plot(fit)
```

```{r}
fit <- lm(gene %>% log2~grade*node,data=kpna2)
plot(fit)
```

When checking the significance of the interaction term, we see that it is significant on the 5% significance level. We therefore keep the full model.

```{r}
library(car)
Anova(fit,type="III")
```

As we are dealing with a factorial design, we can calculate the mean gene expression for each group by the following parameter summations.

```{r}
ExploreModelMatrix::VisualizeDesign(kpna2,~grade*node)$plotlist
```

The researchers want to know the power for testing following hypotheses (remark that we will have to adjust for multiple testing):

- Log fold change between histologic grade 3 and histologic grade 1 for patients with unaffected lymph nodes (=0).

$$H_0: \log_2{FC}_{g3n0-g1n0} = \beta_{g3} = 0$$

- Log fold change between histologic grade 3 and histologic grade 1 for patients with removed lymph nodes (=1). 

$$H_0: \log_2{FC}_{g3n1-g1n1} = \beta_{g3} + \beta_{g3n1} = 0$$

- Log fold change between unaffected and removed lymph nodes for patients of histologic grade 1.

$$H_0: \log_2{FC}_{g1n1-g1n0} = \beta_{n1} = 0$$

- Log fold change between unaffected and removed lymph nodes for patients of histologic grade 3.

$$H_0: \log_2{FC}_{g3n1-g3n0} = \beta_{n1} + \beta_{g3n1} = 0$$

- Difference in log fold change between patients of histological grade 3 and histological grade 1 with removed lymph nodes and log fold change between patients between patients of histological grade 3 and histological grade 1 with unaffected lymph nodes.

$$H_0: \log_2{FC}_{g3n1-g1n1} - \log_2{FC}_{g3n0-g1n0} = \beta_{g3n1} = 0$$ which is an equivalent hypotheses with $$H_0: \log_2{FC}_{g3n1-g3n0} - \log_2{FC}_{g1n1-g1n0} = \beta_{g3n1} = 0$$

We can test this using multcomp, which controls for multiple testing.

```{r}
library(multcomp)

mcp <- glht(fit,linfct = c("grade3 = 0","grade3 + grade3:node1 = 0", "node1 = 0","node1 + grade3:node1 = 0", "grade3:node1 = 0"))
summary(mcp)
```

We get a significant p-value for the first, second, third and fifth hypothesis. The fourth hypothesis is not significant at the overall 5% significance level.

# Power of the tests for each of the contrasts

## Simulation function

Function to simulate data similar to that of our experiment under our model assumptions. 

```{r}
simFastMultipleContrasts <- function(form, data, betas, sd, contrasts, alpha = .05, nSim = 10000, adjust = "holm")
{
    ySim <- rnorm(nrow(data)*nSim,sd=sd)
    dim(ySim) <-c(nrow(data),nSim)
    design <- model.matrix(form, data)
    ySim <- ySim + c(design %*%betas)
    ySim <- t(ySim)
  
    ### Fitting
    fitAll <- limma::lmFit(ySim,design)
  
    ### Inference
    varUnscaled <- t(contrasts)%*%fitAll$cov.coefficients%*%contrasts
    contrasts <- fitAll$coefficients %*%contrasts
    seContrasts <- matrix(diag(varUnscaled)^.5,nrow=nSim,ncol=5,byrow = TRUE)*fitAll$sigma
    tstats <- contrasts/seContrasts
    pvals <- pt(abs(tstats),fitAll$df.residual,lower.tail = FALSE)*2
    pvals <- t(apply(pvals, 1, p.adjust, method = adjust))
    
    return(colMeans(pvals < alpha))
}
```

## power of current experiment

```{r}
nSim <- 20000
betas <- fit$coefficients
sd <- sigma(fit)

contrasts <- matrix(0,nrow=4,ncol=5)
rownames(contrasts) <- names(fit$coefficients)
colnames(contrasts) <- c("graden0","graden1","nodeg1","nodeg3","interaction")
contrasts[2,1] <- 1
contrasts[c(2,4),2] <- 1
contrasts[3,3] <- 1
contrasts[c(3,4),4] <- 1
contrasts[4,5] <- 1

form <- ~ grade*node

power1 <- simFastMultipleContrasts(form, kpna2, betas, sd, contrasts, nSim = nSim)
power1
```

We observe large powers for all contrasts, exept for contrast nodeg3, which has a small effect size. 

## Power for increasing sample size

```{r}
nSim <- 20000
betas <- fit$coefficients
sd <- sigma(fit)

contrasts <- matrix(0,nrow=4,ncol=5)
rownames(contrasts) <- names(fit$coefficients)
colnames(contrasts) <- c("graden0","graden1","nodeg1","nodeg3","interaction")
contrasts[2,1] <- 1
contrasts[c(2,4),2] <- 1
contrasts[3,3] <- 1
contrasts[c(3,4),4] <- 1
contrasts[4,5] <- 1

form <- ~ grade*node

powers <- matrix(NA,nrow=9, ncol=6)
colnames(powers) <- c("n",colnames(contrasts))
powers[,1] <- 2:10

dataAllComb <- data.frame(grade = rep(c(1,3),each=2)%>% as.factor, 
                          node = rep(c(0,1),2)%>%as.factor) 
  
for (i in 1:nrow(powers))
{  
predData <- data.frame(grade = rep(dataAllComb$grade, powers[i,1]), 
                       node = rep(dataAllComb$node, powers[i,1]))
powers[i,-1] <- simFastMultipleContrasts(form, predData, betas, sd, contrasts, nSim = nSim)
}
powers
```

```{r}
powers %>% 
  as.data.frame %>% 
  gather(contrast, power, -n) %>% 
  ggplot(aes(n,power,color=contrast)) +
  geom_line()
```

## Power when FC for grade in patients with unaffected lymph nodes equals 1.5

```{r}
nSim <- 20000
betas2 <- fit$coefficients
betas2["grade3"] <- log2(1.5)
sd <- sigma(fit)

contrasts <- matrix(0,nrow=4,ncol=5)
rownames(contrasts) <- names(fit$coefficients)
colnames(contrasts) <- c("graden0","graden1","nodeg1","nodeg3","interaction")
contrasts[2,1] <- 1
contrasts[c(2,4),2] <- 1
contrasts[3,3] <- 1
contrasts[c(3,4),4] <- 1
contrasts[4,5] <- 1

form <- ~ grade*node

power3 <- simFastMultipleContrasts(form, kpna2, betas2, sd, contrasts,nSim = nSim)
power3
```

We observe a drop in the power for all the contrasts!?

We only would expect to loose power for the contrasts assessing the grade effect because both fold changes have been decreased. 

The reason why all powers are affected is because we also do multiple testing and the multiple testing using the pvalue correction using Holm's method for a particular contrast is also depending on the order of significance of the tests within each simulation. 

When we do both simulations without correcting for multiple testing (adjust = "none") we see that changing the size of the fold change for grade for patients with unaffected lymph nodes only impacts the power for the contrasts for assessing the grade effect. 

```{r}
power1None <- simFastMultipleContrasts(form, kpna2, betas, sd, contrasts, nSim = nSim, adjust = "none")
power3None <- simFastMultipleContrasts(form, kpna2, betas2, sd, contrasts, nSim = nSim, adjust = "none")
power1None
power3None
```





</div>
<div class="footer">
    <hr>
    This work is licensed under the <a href= "https://creativecommons.org/licenses/by-nc-sa/4.0">
    CC BY-NC-SA 4.0</a> licence.
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("08_ExperimentalDesignII_3_kpna2_sol.Rmd");
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
