<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement and Jeroen Gilis" />


<title>Exercise 8.x: Blocking on the estrogen dataset - solution</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>


<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSLS21</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-intro.html">1. Introduction</a>
    </li>
    <li>
      <a href="02-concepts.html">2. Concepts</a>
    </li>
    <li>
      <a href="03-experimentalDesign.html">3. Experimental Design I</a>
    </li>
    <li>
      <a href="04-dataExploration.html">4. Data Exploration</a>
    </li>
    <li>
      <a href="05-statisticalInference.html">5.1 Statistical Inference 1</a>
    </li>
    <li>
      <a href="05-statisticalInference-twosampleT.html">5.2 Statistical Inference 2</a>
    </li>
    <li>
      <a href="06-linearRegression.html">6. Linear Regression</a>
    </li>
    <li>
      <a href="07-Anova.html">7. Analysis of Variance</a>
    </li>
    <li>
      <a href="08-MultipleRegression.html">8. Multiple Regression</a>
    </li>
    <li>
      <a href="08-MultipleRegression-SpecificDesigns.html">8.2 Multiple Regression: Factorial Designs</a>
    </li>
    <li>
      <a href="08-experimentalDesignII.html">8.3. Experimental Design II</a>
    </li>
    <li>
      <a href="08-experimentalDesignIII.html">8.4. Experimental Design II</a>
    </li>
    <li>
      <a href="09-NonparametericStatistics-WilcoxonMannWithney.html">9.1. Nonparametric Statistics 1</a>
    </li>
    <li>
      <a href="09-NonparametericStatistics-KruskalWallis.html">9.2. Nonparametric Statistics 2</a>
    </li>
    <li>
      <a href="10-categoricalDataAnalysis.html">10. Categorical Data Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Labs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04_exercises.html">4. Data Exploration</a>
    </li>
    <li>
      <a href="05_exercises.html">5. Statistical inference</a>
    </li>
    <li>
      <a href="06_exercises.html">6. Simple linear regression</a>
    </li>
    <li>
      <a href="07_exercises.html">7. ANOVA</a>
    </li>
    <li>
      <a href="09_exercises.html">9. Non-parametric statistics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/PSLS21">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Exercise 8.x: Blocking on the estrogen dataset - solution</h1>
<h4 class="author">Lieven Clement and Jeroen Gilis</h4>
<h4 class="date">statOmics, Ghent University (<a href="https://statomics.github.io" class="uri">https://statomics.github.io</a>)</h4>

</div>


<div id="background" class="section level1">
<h1><span class="header-section-number">1</span> Background</h1>
<p>A 2x2 factorial experiment on cancer cell cultures is performed. Two factors were considered in this experiment: estrogen (present or absent) and length of exposure (10 or 48 hours). The study aims to discriminate between early and late response. The researchers assessed gene expression of the TK1 gene (thymidine kinase 1, soluble) a protein-coding gene associated with colorectal cancer.</p>
<p>Note, that the tk1 variable consists of background corrected normalized <span class="math inline">\(\log_2\)</span> transformed intensities, which are a proxy for the <span class="math inline">\(\log_2\)</span> transformed concentration of gene expression product of the TK1 gene.</p>
</div>
<div id="experimental-design" class="section level1">
<h1><span class="header-section-number">2</span> Experimental design</h1>
<ul>
<li><p>There are two explanatory variables: factor estrogen (present and absent) and time (early vs late).</p></li>
<li><p>There response variable is the normalized log2-transformed intensity for the TK1 gene.</p></li>
<li><p>The experimental and observational units are the 8 different cell cultures.</p></li>
<li><p>The treatments are: early-absent, early-present, late-absent and late-present.</p></li>
<li><p>The design is a randomized complete block design.</p></li>
</ul>
</div>
<div id="research-hypotheses" class="section level1">
<h1><span class="header-section-number">3</span> Research hypotheses</h1>
<ul>
<li><p>Is there an early response of the TK1 gene on the estrogen treatment?</p></li>
<li><p>Is there a late response of the TK1 gene on the estrogen treatment?</p></li>
<li><p>Does the response of the TK1 gene on the estrogen treatment change over time?</p></li>
</ul>
<p>Load libraries</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
</div>
<div id="data-import" class="section level1">
<h1><span class="header-section-number">4</span> Data import</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>tk1 &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/PSLS21/data/estrogenTk1.txt&quot;</span>,</span>
<span id="cb2-2"><a href="#cb2-2"></a>                  <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>,</span>
<span id="cb2-3"><a href="#cb2-3"></a>                  <span class="dt">header=</span><span class="ot">TRUE</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="kw">head</span>(tk1)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["estrogen"],"name":[1],"type":["chr"],"align":["left"]},{"label":["time.h"],"name":[2],"type":["int"],"align":["right"]},{"label":["time.f"],"name":[3],"type":["int"],"align":["right"]},{"label":["tk1"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"absent","2":"10","3":"10","4":"8.589664","_rn_":"1"},{"1":"absent","2":"10","3":"10","4":"8.580125","_rn_":"2"},{"1":"present","2":"10","3":"10","4":"11.644443","_rn_":"3"},{"1":"present","2":"10","3":"10","4":"11.752811","_rn_":"4"},{"1":"absent","2":"48","3":"48","4":"7.397465","_rn_":"5"},{"1":"absent","2":"48","3":"48","4":"7.104903","_rn_":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="tidy-data" class="section level1">
<h1><span class="header-section-number">5</span> Tidy data</h1>
<p>We have two columns indicating time. The first <code>time.h</code> should display numeric values, but <code>time.f</code> should display time categories; change <code>time.f</code> to a factor using <code>dplyr</code>. Additionally, change <code>estrogen</code> to a factor.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>tk1 &lt;-<span class="st"> </span>tk1 <span class="op">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">time.f =</span> <span class="kw">as.factor</span>(time.f), <span class="dt">estrogen =</span> <span class="kw">as.factor</span>(estrogen))</span></code></pre></div>
</div>
<div id="data-exploration" class="section level1">
<h1><span class="header-section-number">6</span> Data exploration</h1>
<p>Time and estrogen treatment both may have an effect on the tk1 gene. Moreover, it is also possible that the response on the estrogen treatment changes over time. Therefore, we plot the gene expression for each treatment combination. We do not use boxplots here because there are only two biological repeats for each treatment x time combination. In stead, we make a scatterplot. Note, to avoid overplotting the points, we make use of <code>geom_jitter</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw">set.seed</span>(<span class="dv">4684165</span>) <span class="co"># to make jitter reproducible</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>tk1 <span class="op">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> time.f<span class="op">:</span>estrogen, <span class="dt">y =</span> tk1, <span class="dt">col =</span> time.f<span class="op">:</span>estrogen)) <span class="op">+</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="st">  </span><span class="kw">geom_jitter</span>(<span class="dt">width=</span><span class="fl">0.2</span>) <span class="op">+</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;log2 TK1 intensity&quot;</span>) <span class="op">+</span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;time x estrogen combination&quot;</span>) <span class="op">+</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;TK1 expression in function of time and estrogen&quot;</span>)</span></code></pre></div>
<p><img src="08_x_estrogen_sol_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>The plot suggests</p>
<ol style="list-style-type: decimal">
<li>an estrogen response at both early and late timepoint</li>
<li>an effect of time</li>
<li>a potential change in estrogen response between early and late timepoint.</li>
</ol>
<p>Since there are only two biological repeats for every treatment combination the plot does not allow us to learn much about variability in gene expression for every treatment*time combination.</p>
</div>
<div id="analysis-1-early-response-only" class="section level1">
<h1><span class="header-section-number">7</span> Analysis 1: early response (only)</h1>
<p>Time and estrogen treatment can have an effect on the tk1 gene. Moreover, it is also possible that the response on the estrogen treatment changes over time. Hence, we will have to model the gene expression by using main effects for time, estrogen and a time*estrogen interaction.</p>
<div id="assumptions" class="section level2">
<h2><span class="header-section-number">7.1</span> Assumptions</h2>
<p>List assumptions:</p>
<ol style="list-style-type: decimal">
<li>The observations are independent of each other</li>
<li>Linearity between the response and predictor variable</li>
<li>The residues of the model must be normally distributed</li>
<li>Homoscedasticity of the data</li>
</ol>
<p>The assumption of independence is met, each of the 8 cell cultures was randomly assigned to a time and treatment combination.</p>
<p>For the remaining assumptions, we fit a linear model with a main effects for estrogen dosage, a main effect for time and an estrogen-time interaction effect.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>fit &lt;-<span class="st"> </span><span class="kw">lm</span>(tk1 <span class="op">~</span><span class="st"> </span>estrogen<span class="op">*</span>time.f, <span class="dt">data =</span> tk1)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw">plot</span>(fit)</span></code></pre></div>
<p><img src="08_x_estrogen_sol_files/figure-html/unnamed-chunk-5-1.png" width="672" /><img src="08_x_estrogen_sol_files/figure-html/unnamed-chunk-5-2.png" width="672" /><img src="08_x_estrogen_sol_files/figure-html/unnamed-chunk-5-3.png" width="672" /><img src="08_x_estrogen_sol_files/figure-html/unnamed-chunk-5-4.png" width="672" /></p>
<p>There is not enough data to assess equality of the variance for every treatment x time combination. The QQ-plot of the residuals suggests some deviations from normality, however, there are only a limited number of observations to assess normality.</p>
<p><strong>For demonstrational purposes, we will continue the analysis, but keep in</strong> <strong>mind that we do not have enough data to be sure that the assumptions of the</strong> <strong>model hold! We will come back to this in the <code>conclusions</code> section.</strong></p>
</div>
<div id="interpretation-of-model-parameters-and-statistical-tests" class="section level2">
<h2><span class="header-section-number">7.2</span> Interpretation of model parameters and statistical tests</h2>
<p>We obtain the following estimates for our linear regression model,</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw">summary</span>(fit)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = tk1 ~ estrogen * time.f, data = tk1)
## 
## Residuals:
##         1         2         3         4         5         6         7         8 
##  0.004769 -0.004769 -0.054184  0.054184  0.146281 -0.146281  0.015274 -0.015274 
## 
## Coefficients:
##                          Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)               8.58489    0.07841 109.492 4.17e-08 ***
## estrogenpresent           3.11373    0.11088  28.081 9.57e-06 ***
## time.f48                 -1.33371    0.11088 -12.028 0.000274 ***
## estrogenpresent:time.f48  0.74133    0.15681   4.727 0.009121 ** 
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.1109 on 4 degrees of freedom
## Multiple R-squared:  0.9981, Adjusted R-squared:  0.9967 
## F-statistic: 716.1 on 3 and 4 DF,  p-value: 6.473e-06</code></pre>
<p>and the following corresponding confidence intervals on these estimates.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>CIfit &lt;-<span class="st"> </span><span class="kw">confint</span>(fit)</span>
<span id="cb8-2"><a href="#cb8-2"></a>CIfit</span></code></pre></div>
<pre><code>##                               2.5 %    97.5 %
## (Intercept)               8.3672037  8.802585
## estrogenpresent           2.8058714  3.421594
## time.f48                 -1.6415715 -1.025849
## estrogenpresent:time.f48  0.3059474  1.176710</code></pre>
<p>Next, we should carefully think about the interpretation of these model parameters. We model the log<span class="math inline">\(_2\)</span>-transformed intensities with the following regression model:</p>
<p><span class="math display">\[
y=\beta_0+\beta_{e}x_{e}+\beta_{t}x_{t}+\beta_{et}x_{e}x_{t},
\]</span></p>
<p>with</p>
<ul>
<li><p><span class="math inline">\(\beta_0\)</span> the intercept,</p></li>
<li><p><span class="math inline">\(\beta_{e}\)</span> the main effect for estrogen,</p></li>
<li><p><span class="math inline">\(x_{e}\)</span> a dummy variable for estrogen which is 0 for the control treatment in the absence of estrogen and 1 for the treatment with estrogen,</p></li>
<li><p><span class="math inline">\(\beta_{t}\)</span> the main effect for time,</p></li>
<li><p><span class="math inline">\(x_{t}\)</span> a dummy variable that is 0 for the measurements at the early time (10h) and 1 for measurements at the late timepoint (48h)</p></li>
<li><p><span class="math inline">\(\beta_{et}\)</span> the interaction effect between estrogen and time.</p></li>
</ul>
<p>To ease the interpretation of the parameters, <span class="math inline">\(\log_2\)</span> transformed mean intensities are given for each treatment group as well as corresponding contrasts between treatments, which have an interpretation in terms of <span class="math inline">\(\log_2\)</span> transformed fold changes (FC).</p>
<ul>
<li><p><span class="math inline">\(\log_2\hat \mu_{A10}=\hat\beta_0\)</span>, <span class="math inline">\(\log_2 \hat\mu_{P10}=\hat\beta_0+\hat\beta_e\)</span> –&gt; <span class="math inline">\(\log_2 \widehat{FC}_{P10-A10}=\hat\beta_e\)</span></p></li>
<li><p><span class="math inline">\(\log_2 \hat \mu_{A48}=\hat\beta_0+\hat\beta_t\)</span>, <span class="math inline">\(\log_2 \hat \mu_{P48}=\hat\beta_0+\hat\beta_e+\hat\beta_t+\hat\beta_{et}\)</span> –&gt; <span class="math inline">\(\log_2 \widehat{FC}_{P48-A48}=\hat \beta_e +\hat\beta_{et}\)</span></p></li>
<li><p><span class="math inline">\(\log_2 \widehat{FC}_{A48-A10}=\hat\beta_{t}\)</span>, <span class="math inline">\(\log_2 \widehat{FC}_{P48-P10}=\hat\beta_{t}+\hat\beta_{et}\)</span> –&gt; <span class="math inline">\(\log_2\frac{\widehat{FC}_{P48-A48}}{\widehat{FC}_{P10-A10}}=\hat\beta_{et}\)</span></p></li>
</ul>
<p>with <span class="math inline">\(\log_2\hat \mu_{A10}\)</span>, <span class="math inline">\(\log_2\hat \mu_{P10}\)</span>, <span class="math inline">\(\log_2\hat \mu_{A48}\)</span> and <span class="math inline">\(\log_2\hat \mu_{P48}\)</span> the estimated mean <span class="math inline">\(\log_2\)</span> transformed intensity for the control treatment in the absence of estrogen at 10h, the estrogen treatment at 10h, the control treatment at 48h and the estrogen treatment at 48h, respectively. With <span class="math inline">\(\log_2 \widehat{FC}_{b-a}\)</span> we indicate <span class="math inline">\(\log_2\)</span> transformed fold change estimates between treatment B and treatment A, i.e. <span class="math inline">\(\log_2 \widehat{FC}_{b-a}=\log_2 \hat\mu_{b}-\log_2 \hat\mu_a=\log_2 \frac{\hat\mu_{b}}{\hat\mu_{a}}\)</span>.</p>
<p>The model immediately provides statistical tests for assessing the significance of fold changes between the estrogen and control treatment at the early time point <span class="math inline">\(\log_2 \widehat{FC}_{P10-A10}\)</span>, fold changes between the control at the late time point and the control at the early time point <span class="math inline">\(\log_2 \widehat{FC}_{A48-P10}\)</span> and for changes in fold change between control and estrogen treatment at the late and the early timepoint <span class="math inline">\(\log_2\frac{\widehat{FC}_{P48-A48}}{\widehat{FC}_{P10-A10}}\)</span>, the interaction term.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">summary</span>(fit)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = tk1 ~ estrogen * time.f, data = tk1)
## 
## Residuals:
##         1         2         3         4         5         6         7         8 
##  0.004769 -0.004769 -0.054184  0.054184  0.146281 -0.146281  0.015274 -0.015274 
## 
## Coefficients:
##                          Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)               8.58489    0.07841 109.492 4.17e-08 ***
## estrogenpresent           3.11373    0.11088  28.081 9.57e-06 ***
## time.f48                 -1.33371    0.11088 -12.028 0.000274 ***
## estrogenpresent:time.f48  0.74133    0.15681   4.727 0.009121 ** 
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.1109 on 4 degrees of freedom
## Multiple R-squared:  0.9981, Adjusted R-squared:  0.9967 
## F-statistic: 716.1 on 3 and 4 DF,  p-value: 6.473e-06</code></pre>
<p>For communicating the results, we may transform the model parameters to the original scale, i.e., the (geometric) average intensity in stead of the average log2 intensity, using:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co"># Transform parameters and the CI back to the original scale</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="dv">2</span><span class="op">^</span>fit<span class="op">$</span>coef</span></code></pre></div>
<pre><code>##              (Intercept)          estrogenpresent                 time.f48 
##              383.9818884                8.6561923                0.3967466 
## estrogenpresent:time.f48 
##                1.6717150</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="dv">2</span><span class="op">^</span>CIfit</span></code></pre></div>
<pre><code>##                                2.5 %      97.5 %
## (Intercept)              330.2016989 446.5212961
## estrogenpresent            6.9928055  10.7152508
## time.f48                   0.3205072   0.4911211
## estrogenpresent:time.f48   1.2362302   2.2606072</code></pre>
<ul>
<li><p>The geometric average intensity for the control treatment on the early timepoint equals <span class="math inline">\(\exp(\hat \beta_0)\)</span>=383.98.</p></li>
<li><p>The estrogen effect at the early time point and the time effect are extremely significant (p&lt;&lt;0.001).</p></li>
<li><p>The expression of the tk1 gene at the early timepoint is on average 8.66 times higher after estrogen treatment (95% CI [6.99, 10.72]).</p></li>
<li><p>The gene expression in the control treatment is on average 2.52 times lower (or 0.4 times higher) at the late time point (95% CI [3.12, 2.04]).</p></li>
<li><p>The difference in estrogen response at the late timepoint and the early timepoint is strongly significant (p=0.009). The fold change due to the estrogen treatment is on average 1.67 times higher at the late time point (95% CI [1.24, 2.26]).</p></li>
</ul>
</div>
</div>
<div id="analysis-2-late-response-only" class="section level1">
<h1><span class="header-section-number">8</span> Analysis 2: Late response (only)</h1>
<p>The researchers, however, are also interested in testing the significance of the fold change between estrogen and control treatment at the late timepoint. This involves assessing a contrast of the model parameters: <span class="math inline">\(\log_2 \widehat{FC}_{P48-A48}=\hat \beta_e +\hat\beta_{et}\)</span>. We can test this linear combination of model parameters using the <code>multcomp</code> package.</p>
<div id="test" class="section level2">
<h2><span class="header-section-number">8.1</span> Test</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="kw">library</span>(multcomp)</span>
<span id="cb16-2"><a href="#cb16-2"></a></span>
<span id="cb16-3"><a href="#cb16-3"></a>testFCP48_A48 &lt;-<span class="st"> </span><span class="kw">glht</span>(<span class="dt">model =</span> fit, </span>
<span id="cb16-4"><a href="#cb16-4"></a>                      <span class="dt">linfct =</span> <span class="st">&quot;estrogenpresent + estrogenpresent:time.f48 = 0&quot;</span>)</span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="kw">summary</span>(testFCP48_A48)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: lm(formula = tk1 ~ estrogen * time.f, data = tk1)
## 
## Linear Hypotheses:
##                                                 Estimate Std. Error t value
## estrogenpresent + estrogenpresent:time.f48 == 0   3.8551     0.1109   34.77
##                                                 Pr(&gt;|t|)    
## estrogenpresent + estrogenpresent:time.f48 == 0 4.08e-06 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<div id="confidence-intervals-fc-at-time-48h." class="section level3">
<h3><span class="header-section-number">8.1.1</span> Confidence intervals FC at time 48h.</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># CI for fold changes on time point 48h</span></span>
<span id="cb18-2"><a href="#cb18-2"></a>CIFCP48_A48 &lt;-<span class="st"> </span><span class="kw">confint</span>(testFCP48_A48)</span>
<span id="cb18-3"><a href="#cb18-3"></a>CIFCP48_A48</span></code></pre></div>
<pre><code>## 
##   Simultaneous Confidence Intervals
## 
## Fit: lm(formula = tk1 ~ estrogen * time.f, data = tk1)
## 
## Quantile = 2.7764
## 95% family-wise confidence level
##  
## 
## Linear Hypotheses:
##                                                 Estimate lwr    upr   
## estrogenpresent + estrogenpresent:time.f48 == 0 3.8551   3.5472 4.1629</code></pre>
<p>For communicating the results, we may transform the model parameters to the original scale, i.e., the (geometric) average intensity in stead of the average log2 intensity, using:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="co"># Original scale</span></span>
<span id="cb20-2"><a href="#cb20-2"></a><span class="dv">2</span><span class="op">^</span>CIFCP48_A48<span class="op">$</span>confint</span></code></pre></div>
<pre><code>##                                            Estimate      lwr      upr
## estrogenpresent + estrogenpresent:time.f48 14.47069 11.68998 17.91285
## attr(,&quot;conf.level&quot;)
## [1] 0.95
## attr(,&quot;calpha&quot;)
## [1] 2.776445</code></pre>
<p>The estrogen effect at the late time point is also extremely significant (p&lt;&lt;0.001). The expression of the tk1 gene at the late timepoint is on average 14.47 times higher after estrogen treatment (95% CI [11.69, 17.91]).</p>
</div>
</div>
</div>
<div id="analysis-3-simultaneously-assess-all-research-hypotheses" class="section level1">
<h1><span class="header-section-number">9</span> Analysis 3: simultaneously assess all research hypotheses</h1>
<p>Alternatively, we can simultaneously assess all research hypotheses for our linear regression model using the package <code>multcomp</code>. This is the most elegant way for addressing all research hypotheses, and additionally allows for efficient multiple testing correction.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>testFCAll &lt;-<span class="st"> </span><span class="kw">glht</span>(<span class="dt">model =</span> fit, </span>
<span id="cb22-2"><a href="#cb22-2"></a>                  <span class="dt">linfct =</span> <span class="kw">c</span>(<span class="st">&quot;estrogenpresent = 0&quot;</span>,</span>
<span id="cb22-3"><a href="#cb22-3"></a>                             <span class="st">&quot;estrogenpresent + estrogenpresent:time.f48 = 0&quot;</span>,</span>
<span id="cb22-4"><a href="#cb22-4"></a>                             <span class="st">&quot;estrogenpresent:time.f48 = 0&quot;</span>))</span>
<span id="cb22-5"><a href="#cb22-5"></a><span class="kw">summary</span>(testFCAll)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: lm(formula = tk1 ~ estrogen * time.f, data = tk1)
## 
## Linear Hypotheses:
##                                                 Estimate Std. Error t value
## estrogenpresent == 0                              3.1137     0.1109  28.081
## estrogenpresent + estrogenpresent:time.f48 == 0   3.8551     0.1109  34.767
## estrogenpresent:time.f48 == 0                     0.7413     0.1568   4.727
##                                                 Pr(&gt;|t|)    
## estrogenpresent == 0                              &lt;0.001 ***
## estrogenpresent + estrogenpresent:time.f48 == 0   &lt;0.001 ***
## estrogenpresent:time.f48 == 0                     0.0188 *  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="co"># Confidence intervals </span></span>
<span id="cb24-2"><a href="#cb24-2"></a>CIFCAll &lt;-<span class="st"> </span><span class="kw">confint</span>(testFCAll)</span>
<span id="cb24-3"><a href="#cb24-3"></a>CIFCAll</span></code></pre></div>
<pre><code>## 
##   Simultaneous Confidence Intervals
## 
## Fit: lm(formula = tk1 ~ estrogen * time.f, data = tk1)
## 
## Quantile = 3.5137
## 95% family-wise confidence level
##  
## 
## Linear Hypotheses:
##                                                 Estimate lwr    upr   
## estrogenpresent == 0                            3.1137   2.7241 3.5033
## estrogenpresent + estrogenpresent:time.f48 == 0 3.8551   3.4654 4.2447
## estrogenpresent:time.f48 == 0                   0.7413   0.1903 1.2923</code></pre>
<p>For communicating the results, we may transform the model parameters to the original scale, i.e., the (geometric) average intensity in stead of the average log2 intensity, using:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="co"># Confidence intervals on original scale</span></span>
<span id="cb26-2"><a href="#cb26-2"></a><span class="dv">2</span><span class="op">^</span>CIFCAll<span class="op">$</span>confint</span></code></pre></div>
<pre><code>##                                             Estimate       lwr       upr
## estrogenpresent                             8.656192  6.607557 11.339996
## estrogenpresent + estrogenpresent:time.f48 14.470686 11.045951 18.957241
## estrogenpresent:time.f48                    1.671715  1.141024  2.449231
## attr(,&quot;conf.level&quot;)
## [1] 0.95
## attr(,&quot;calpha&quot;)
## [1] 3.513748</code></pre>
</div>
<div id="conclusions" class="section level1">
<h1><span class="header-section-number">10</span> Conclusions</h1>
<p>There was an extremely significant effect of estrogen on the expression of the TK1 gene 10h and 48h after estrogen dosage ( <span class="math inline">\(p&lt;&lt;0.001\)</span>).</p>
<p>The TK1 gene was significantly upregulated after estrogen treatment as compared to a control treatment without estrogen. On average, the fold change between the estrogen treatment and control varied form 14.47 (95% CI [6.61, 11.34]) 10 hours after estrogen dosage up to 14.47 (95% CI [11.05, 18.96]) at the late timepoint (48h), respectively.</p>
<p>The increase in fold change between the early and the late timepoint was significant (<span class="math inline">\(p=0.02\)</span>), fold change increase: 1.67, 95% CI [1.14, 2.45]).</p>
<p>All p-values and confidence intervals were adjusted for multiple testing.</p>
</div>

<div id="rmd-source-code">---
title: "Exercise 8.x: Blocking on the estrogen dataset - solution"   
author: "Lieven Clement and Jeroen Gilis"
date: "statOmics, Ghent University (https://statomics.github.io)"  
output:
    html_document:
      code_download: true    
      theme: cosmo
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
---

# Background

A 2x2 factorial experiment on cancer cell cultures is performed. Two factors 
were considered in this experiment: estrogen (present or absent) and length of 
exposure (10 or 48 hours). The study aims to discriminate between early and 
late response. The researchers assessed gene expression of the TK1 gene 
(thymidine kinase 1, soluble) a protein-coding gene associated with colorectal 
cancer. 

Note, that the tk1 variable consists of background corrected normalized 
$\log_2$ transformed intensities, which are a proxy for the $\log_2$ transformed
concentration of gene expression product of the TK1 gene. 

# Experimental design 

- There are two explanatory variables: factor estrogen (present and absent) and
time (early vs late). 

- There response variable is the normalized log2-transformed intensity for 
the TK1 gene.

- The experimental and observational units are the 8 different cell cultures.

- The treatments are: early-absent, early-present, late-absent and late-present.

- The design is a randomized complete block design.

# Research hypotheses

- Is there an early response of the TK1 gene on the estrogen treatment?

- Is there a late response of the TK1 gene on the estrogen treatment?

- Does the response of the TK1 gene on the estrogen treatment change over time? 

Load libraries

```{r, message=FALSE, warning=FALSE}
library(tidyverse)
```

# Data import

```{r}
tk1 <- read.table("https://raw.githubusercontent.com/statOmics/PSLS21/data/estrogenTk1.txt",
                  sep=",",
                  header=TRUE)
head(tk1)
```

# Tidy data

We have two columns indicating time. The first `time.h` should display numeric
values, but `time.f` should display time categories; change `time.f` to a
factor using `dplyr`. Additionally, change `estrogen` to a factor.

```{r}
tk1 <- tk1 %>%
    mutate(time.f = as.factor(time.f), estrogen = as.factor(estrogen))
```

# Data exploration

Time and estrogen treatment both may have an effect on the tk1 gene. Moreover, 
it is also possible that the response on the estrogen treatment changes over 
time. Therefore, we plot the gene expression for each treatment combination. 
We do not use boxplots here because there are only two biological repeats for 
each treatment x time combination. In stead, we make a scatterplot. Note, to
avoid overplotting the points, we make use of `geom_jitter`.

```{r}
set.seed(4684165) # to make jitter reproducible
tk1 %>%
  ggplot(aes(x = time.f:estrogen, y = tk1, col = time.f:estrogen)) +
  geom_jitter(width=0.2) +
  theme_bw() +
  ylab("log2 TK1 intensity") +
  xlab("time x estrogen combination") +
  ggtitle("TK1 expression in function of time and estrogen")
```

The plot suggests

1. an estrogen response at both early and late timepoint
2. an effect of time
3. a potential change in estrogen response between early and late timepoint.

Since there are only two biological repeats for every treatment combination 
the plot does not allow us to learn much about variability in gene expression 
for every treatment*time combination. 

# Analysis 1: early response (only)

Time and estrogen treatment can have an effect on the tk1 gene. Moreover, it is
also possible that the response on the estrogen treatment changes over time. 
Hence, we will have to model the gene expression by using main effects for time,
estrogen and a time*estrogen interaction. 

## Assumptions

List assumptions:

1. The observations are independent of each other
2. Linearity between the response and predictor variable
3. The residues of the model must be normally distributed
4. Homoscedasticity of the data

The assumption of independence is met, each of the 8 cell cultures was randomly
assigned to a time and treatment combination.

For the remaining assumptions, we fit a linear model with a main effects for 
estrogen dosage, a main effect for time and an estrogen-time interaction effect.
 
```{r}
fit <- lm(tk1 ~ estrogen*time.f, data = tk1)
plot(fit)
```

There is not enough data to assess equality of the variance for every 
treatment x time combination. The QQ-plot of the residuals suggests some 
deviations from normality, however, there are only a limited number of 
observations to assess normality. 

**For demonstrational purposes, we will continue the analysis, but keep in**
**mind that we do not have enough data to be sure that the assumptions of the**
**model hold! We will come back to this in the `conclusions` section.**
 
## Interpretation of model parameters and statistical tests

We obtain the following estimates for our linear regression model,

```{r}
summary(fit)
```

and the following corresponding confidence intervals on these estimates.

```{r}
CIfit <- confint(fit)
CIfit
```

Next, we should carefully think about the interpretation of these model 
parameters. We model the log$_2$-transformed intensities with the following 
regression model: 

$$
y=\beta_0+\beta_{e}x_{e}+\beta_{t}x_{t}+\beta_{et}x_{e}x_{t},
$$

with 

- $\beta_0$ the intercept,

- $\beta_{e}$ the main effect for estrogen, 

- $x_{e}$ a dummy variable for estrogen which is 0 for the control treatment in 
the absence of estrogen and 1 for the treatment with estrogen, 

- $\beta_{t}$ the main effect for time, 

- $x_{t}$ a dummy variable that is 0 for the measurements at the early time 
(10h) and 1 for measurements at the late timepoint (48h) 

- $\beta_{et}$ the interaction effect between estrogen and time. 

To ease the interpretation of the parameters, $\log_2$ transformed mean 
intensities are given for each treatment group as well as corresponding 
contrasts between treatments, which have an interpretation in terms of $\log_2$ 
transformed fold changes (FC). 

- $\log_2\hat \mu_{A10}=\hat\beta_0$, $\log_2 \hat\mu_{P10}=\hat\beta_0+\hat\beta_e$ 
-->   $\log_2 \widehat{FC}_{P10-A10}=\hat\beta_e$

- $\log_2 \hat \mu_{A48}=\hat\beta_0+\hat\beta_t$, $\log_2 \hat \mu_{P48}=\hat\beta_0+\hat\beta_e+\hat\beta_t+\hat\beta_{et}$ 
-->   $\log_2 \widehat{FC}_{P48-A48}=\hat \beta_e +\hat\beta_{et}$

- $\log_2 \widehat{FC}_{A48-A10}=\hat\beta_{t}$, $\log_2 \widehat{FC}_{P48-P10}=\hat\beta_{t}+\hat\beta_{et}$ 
-->   $\log_2\frac{\widehat{FC}_{P48-A48}}{\widehat{FC}_{P10-A10}}=\hat\beta_{et}$

with $\log_2\hat \mu_{A10}$, $\log_2\hat \mu_{P10}$, $\log_2\hat \mu_{A48}$ 
and $\log_2\hat \mu_{P48}$ the estimated mean $\log_2$ transformed intensity 
for the control treatment in the absence of estrogen at 10h, the estrogen 
treatment at 10h, the control treatment at 48h and the estrogen treatment 
at 48h, respectively. With $\log_2 \widehat{FC}_{b-a}$ we indicate $\log_2$ 
transformed fold change estimates between treatment B and treatment A, 
i.e. $\log_2 \widehat{FC}_{b-a}=\log_2 \hat\mu_{b}-\log_2 \hat\mu_a=\log_2 \frac{\hat\mu_{b}}{\hat\mu_{a}}$.

The model immediately provides statistical tests for assessing the significance 
of fold changes between the estrogen and control treatment at the early time 
point $\log_2 \widehat{FC}_{P10-A10}$, fold changes between the control at 
the late time point and the control at the early time point 
$\log_2 \widehat{FC}_{A48-P10}$ and for changes in fold change between 
control and estrogen treatment at the late and the early timepoint $\log_2\frac{\widehat{FC}_{P48-A48}}{\widehat{FC}_{P10-A10}}$, the interaction 
term. 

```{r}
summary(fit)
```

For communicating the results, we may transform the model parameters to the
original scale, i.e., the (geometric) average intensity in stead of the average 
log2 intensity, using:

```{r}
# Transform parameters and the CI back to the original scale
2^fit$coef
2^CIfit
```

- The geometric average intensity for the control treatment on the early 
timepoint equals $\exp(\hat \beta_0)$=`r round(2^fit$coef["(Intercept)"],2)`.

- The estrogen effect at the early time point and the time effect are extremely 
significant (p<<0.001). 

- The expression of the tk1 gene at the early timepoint is on average 
`r round(2^fit$coef["estrogenpresent"],2)` times higher after estrogen treatment 
(95% CI [`r round(2^CIfit["estrogenpresent",],2)`]). 

- The gene expression in the control treatment is on average 
`r round(1/2^fit$coef["time.f48"],2)` times lower (or 
`r round(2^fit$coef["time.f48"],2)` times higher) at the late time point 
(95% CI [`r round(1/2^CIfit["time.f48",],2)`]).

- The difference in estrogen response at the late timepoint and the early 
timepoint is strongly significant (p=0.009). The fold change due to the estrogen
treatment is on average `r round(2^fit$coef["estrogenpresent:time.f48"],2)` 
times higher at the late time point 
(95% CI [`r round(2^CIfit["estrogenpresent:time.f48",],2)`]).

# Analysis 2: Late response (only)

The researchers, however, are also interested in testing the significance of
the fold change between estrogen and control treatment at the late timepoint.
This involves assessing a contrast of the model parameters:
$\log_2 \widehat{FC}_{P48-A48}=\hat \beta_e +\hat\beta_{et}$. We can test this 
linear combination of model parameters using the `multcomp` package.

## Test

```{r, message=FALSE, warning=FALSE}
library(multcomp)

testFCP48_A48 <- glht(model = fit, 
                      linfct = "estrogenpresent + estrogenpresent:time.f48 = 0")
summary(testFCP48_A48)
```

### Confidence intervals FC at time 48h.

```{r}
# CI for fold changes on time point 48h
CIFCP48_A48 <- confint(testFCP48_A48)
CIFCP48_A48
```

For communicating the results, we may transform the model parameters to the
original scale, i.e., the (geometric) average intensity in stead of the average 
log2 intensity, using:

```{r}
# Original scale
2^CIFCP48_A48$confint
```

The estrogen effect at the late time point is also extremely significant 
(p<<0.001). The expression of the tk1 gene at the late timepoint is on average 
`r round(2^CIFCP48_A48$confint,2)[1]` times higher after estrogen treatment 
(95% CI [`r round(2^CIFCP48_A48$confint,2)[-1]`]). 

# Analysis 3: simultaneously assess all research hypotheses

Alternatively, we can simultaneously assess all research hypotheses for our
linear regression model using the package `multcomp`. This is the most elegant
way for addressing all research hypotheses, and additionally allows for 
efficient multiple testing correction.

```{r}
testFCAll <- glht(model = fit, 
                  linfct = c("estrogenpresent = 0",
                             "estrogenpresent + estrogenpresent:time.f48 = 0",
                             "estrogenpresent:time.f48 = 0"))
summary(testFCAll)
```

```{r}
# Confidence intervals 
CIFCAll <- confint(testFCAll)
CIFCAll
```

For communicating the results, we may transform the model parameters to the
original scale, i.e., the (geometric) average intensity in stead of the average 
log2 intensity, using:

```{r}
# Confidence intervals on original scale
2^CIFCAll$confint
```

# Conclusions

There was an extremely significant effect of estrogen on the expression of the 
TK1 gene 10h and 48h after estrogen dosage ( $p<<0.001$). 

The TK1 gene was significantly upregulated after estrogen treatment as compared
to a control treatment without estrogen. On average, the fold change between 
the estrogen treatment and control varied form 
`r round(2^CIFCAll$confint[2,1],2)` 
(95% CI [`r round(2^CIFCAll$confint[1,2:3],2)`]) 10 hours after 
estrogen dosage up to `r round(2^CIFCAll$confint[2,1],2)` 
(95% CI [`r round(2^CIFCAll$confint[2,2:3],2)`]) at the late timepoint 
(48h), respectively.  

The increase in fold change between the early and the late timepoint was 
significant ($p=0.02$), fold change increase: 
`r round(2^CIFCAll$confint[3,1],2)`, 
95% CI [`r round(2^CIFCAll$confint[3,2:3],2)`]). 

All p-values and confidence intervals were adjusted for multiple testing. 


</div>
<div class="footer">
    <hr>
    This work is licensed under the <a href= "https://creativecommons.org/licenses/by-nc-sa/4.0">
    CC BY-NC-SA 4.0</a> licence.
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("08_x_estrogen_sol.Rmd");
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
